pkg/controller/installconfig/installconfig_controller.go

```diff
@@ -0,0 +1,211 @@
+	return &ReconcileInstallConfig{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("installconfig-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource InstallConfig
+	err = c.Watch(&source.Kind{Type: &v1alpha1.InstallConfig{}}, &handler.EnqueueRequestForObject{}, predicate.GenerationChangedPredicate{})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// blank assignment to verify that ReconcileInstallConfig implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileInstallConfig{}
+
+// ReconcileInstallConfig reconciles a InstallConfig object
+type ReconcileInstallConfig struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a InstallConfig object and makes changes based on the state read
+// and what is in the InstallConfig.Spec
+// Note: The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileInstallConfig) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling InstallConfig")
+
+	// Fetch the InstallConfig instance
+	installConfig := &v1alpha1.InstallConfig{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, installConfig)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	if err := r.EnsureCheSubscription(reqLogger, installConfig); err != nil {
+		reqLogger.Error(err, "failed to create subscription for che", "Che.Namespace", installConfig.Spec.CheOperatorSpec.Namespace)
+		return r.StatusUpdate(reqLogger, installConfig, r.setStatusCheSubscriptionFailed, err.Error())
+	}
+
+	return r.StatusUpdate(reqLogger, installConfig, r.setStatusCheSubscriptionCreated, "che operator subscription created")
+}
+
+func (r *ReconcileInstallConfig) EnsureCheSubscription(logger logr.Logger, installConfig *v1alpha1.InstallConfig) error {
+	ns, err := r.ensureCheNamespace(logger, installConfig)
+	if err != nil {
+		return err
+	}
+
+	if err := r.createCheOperatorGroup(logger, ns, installConfig); err != nil {
```

Comment:
Same here. We should wait for the operator group to be created before trying to create the subscription, right? So, I guess we need to watch for namespaces, operator groups and subscriptions as secondary resources and do one thing at a time.

Reply:
As our operator is not going to create og and subscription just once, I don't think we should watch it. How about using `reconcile.Result{RequeueAfter: 1 * time.Second}`? This way we'll be sure about the existence of require object. wdyt?

Reply:
If the subscription or group were deleted, do we want to recreate them? If so then we better watch.
But let's have something simple for now. We can improve it later.

Reply:
With current implementation, we'll be just creating ns/og/subscription once, after that if user deletes anyone of them, operator is not going to maintain it's state. I think we should clear this with PM